compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java

```diff
@@ -1265,9 +1265,16 @@ public void visit(BLangClassDefinition classDefinition) {
         lookupNodes(classDefinition.referencedFields);
         lookupNode(classDefinition.initFunction);
         lookupNodes(classDefinition.functions);
+        lookupAnnAttachmentsAttachedToMethods(classDefinition.functions);
```

Comment:
Is this a temporary workaround for the position issue?

Reply:
Seems like that

Reply:
This is a different approach, since extending the function position to have annot-attachments is only required here, I think it is better to handle it here.

Reply:
So this doesn't quite fix https://github.com/ballerina-platform/ballerina-lang/issues/38053, right?

Reply:
I think this is the correct fix. We faced the same kind of issue in NodeFinder.java. At that time we decided to solve that in NodeFinder.java
Please check https://github.com/ballerina-platform/ballerina-lang/issues/36731#issuecomment-1216448160